apps/typegpu-docs/src/content/examples/simple/color/index.ts

```diff
@@ -0,0 +1,274 @@
+import {
+  gamutClipAdaptiveL0cusp,
+  gamutClipAdaptiveL05,
+  linearToSrgb,
+  oklabToLinearRgb,
+} from '@typegpu/color';
+import tgpu from 'typegpu';
+import * as d from 'typegpu/data';
+
+const cssProbePosition = d.vec2f(0.5, 0.5);
+
+const presentationFormat = navigator.gpu.getPreferredCanvasFormat();
+const canvas = document.querySelector('canvas') as HTMLCanvasElement;
+const cssProbe = document.querySelector('#css-probe') as HTMLDivElement;
+const probePositionText = document.querySelector(
+  '#probe-position',
+) as HTMLDivElement;
+canvas.parentElement?.appendChild(cssProbe);
+canvas.parentElement?.appendChild(probePositionText);
+const context = canvas.getContext('webgpu') as GPUCanvasContext;
+
+const cleanupController = new AbortController();
+
+document.addEventListener(
+  'mousemove',
+  (ev) => {
+    const rect = canvas.getBoundingClientRect();
+    cssProbePosition.x = (ev.clientX - rect.x) / rect.width;
+    cssProbePosition.y = 1 - (ev.clientY - rect.y) / rect.height;
+    draw();
+  },
+  {
+    signal: cleanupController.signal,
+  },
+);
+
+const root = await tgpu.init();
+
+context.configure({
+  device: root.device,
+  format: presentationFormat,
+  alphaMode: 'premultiplied',
+});
+
+const mainVertex = tgpu['~unstable'].vertexFn({
+  in: { vertexIndex: d.builtin.vertexIndex },
+  out: { outPos: d.builtin.position, uv: d.vec2f },
+}) /* wgsl */`{
+    var pos = array<vec2f, 3>(
+      vec2(-1, -1),
+      vec2(3, -1),
+      vec2(-1, 3)
+    );
+
+    return Out(vec4f(pos[in.vertexIndex], 0.0, 1.0), pos[in.vertexIndex]);
+  }`;
+
+const Uniforms = d.struct({
+  hue: d.f32,
+  alpha: d.f32,
+});
+
+const uniforms = Uniforms({
+  hue: 0.7,
+  alpha: 0.05,
+});
+
+const uniformsBindGroupLayout = tgpu.bindGroupLayout({
+  uniforms: { uniform: Uniforms },
+});
+
+const uniformsBuffer = root.createBuffer(Uniforms, uniforms).$usage('uniform');
+const uniformsBindGroup = root.createBindGroup(uniformsBindGroupLayout, {
+  uniforms: uniformsBuffer,
+});
+
+const modulo = tgpu['~unstable'].fn(
```

Comment:
this could also be added to `std`. I really hate that neither `%` and `mod` are basically useless https://github.com/gpuweb/gpuweb/issues/3987

Reply:
maybe not `std` as it is not in actual WGSL, but I feel like this is going to be a very common function for me to copy paste around.